packages/api/test/models/activityLog.test.ts

```diff
@@ -89,4 +89,88 @@ describe("ActivityLog Model", function () {
             "The message attribute should have a maximum of 512 characters.",
         );
     });
+
+    it("should not be created when component is undefined", async () => {
+        const newLog = new ActivityLogModel({
+            ...logTemplate,
+            component: undefined,
+        });
+
+        await assertThrowsAsync(
+            async () => newLog.save(),
+            "The component attribute is required.",
+        );
+    });
+
+    it("should not be created when component is null", async () => {
+        const newLog = new ActivityLogModel({
+            ...logTemplate,
+            component: null,
+        });
+
+        await assertThrowsAsync(
+            async () => newLog.save(),
+            "The component attribute is required.",
+        );
+    });
+
+    it("should not be created when component is empty", async () => {
+        const newLog = new ActivityLogModel({
+            ...logTemplate,
+            component: "",
+        });
+
+        await assertThrowsAsync(
+            async () => newLog.save(),
+            "The component attribute is required.",
+        );
+    });
+
+    it("should not be created when component is invalid", async () => {
+        const newLog = new ActivityLogModel({
+            ...logTemplate,
+            component: "no-api",
+        });
+
+        await assertThrowsAsync(
+            async () => newLog.save(),
+            "The component attribute is not from defined origins.",
+        );
+    });
+
+    it("should be created when context is null", async () => {
+        const newLog = new ActivityLogModel({
+            ...logTemplate,
+            context: null,
+        });
+        await newLog.save();
+        assert.isFalse(
+            newLog.isNew,
+            "The log should be persisted to the database.",
+        );
+    });
+
+    it("should be created when context is undefined", async () => {
+        const newLog = new ActivityLogModel({
+            ...logTemplate,
+            context: undefined,
+        });
+        await newLog.save();
+        assert.isFalse(
+            newLog.isNew,
+            "The log should be persisted to the database.",
+        );
+    });
+
+    it("should be created if createdAt is valid", async () => {
+        const newLog = new ActivityLogModel({ ...logTemplate });
+        await newLog.save();
+        assert.typeOf(newLog.createdAt, "date", "This is a valid date");
```

Comment:
"This is a valid date"

The message that you specify to an assertion should say what went wrong. Read the docs.

The correct message is:

"The createdAt attribute should be assigned by default."